spec/features/users/index_users_spec.rb

```diff
@@ -51,6 +51,23 @@
       expect(page).to be_accessible.within '#main-content'
     end
 
+    context 'when searching' do
+      let(:user) { create(:user, :with_manager_role, first_name: 'John') }
+      let!(:other_user) { create(:user, :with_caseworker_role, first_name: 'Jane') }
+
+      scenario 'I search' do
+        click_link_or_button 'Manage users'
+        click_button 'Show filters'
+        fill_in 'Name, username or email', with: user.first_name
+        click_on 'Apply filters'
+        expect(page).to have_content(user.email)
+        expect(page).to have_no_content(other_user.email)
```

Comment:
nit: to make it more readable, add an empty line between an expectation and next action.
In this case between:
`expect(page).to have_no_content(other_user.email)` and
`click_on 'Clear filters'`